packages/front-end/components/Settings/EditDataSource/DataSourceExperimentProperties/DataSourceEditExperimentEventPropertiesModal.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC } from "react";
+import { useForm } from "react-hook-form";
+import {
+  DataSourceEvents,
+  DataSourceInterfaceWithParams,
+} from "back-end/types/datasource";
+import Field from "../../../Forms/Field";
+import Modal from "../../../Modal";
+
+type DataSourceEditExperimentEventPropertiesProps = {
+  dataSource: DataSourceInterfaceWithParams;
+  onSave: (experimentProps: DataSourceEvents) => void;
+  onCancel: () => void;
+};
+
+export const DataSourceEditExperimentEventPropertiesModal: FC<DataSourceEditExperimentEventPropertiesProps> = ({
+  dataSource,
+  onCancel,
+  onSave,
+}) => {
+  const form = useForm<DataSourceEvents>({
+    defaultValues: {
+      experimentEvent: dataSource.settings?.events?.experimentEvent || "",
+      experimentIdProperty:
+        dataSource.settings?.events?.experimentIdProperty || "",
+      variationIdProperty:
+        dataSource.settings?.events?.variationIdProperty || "",
+    },
+  });
+
+  const handleSubmit = form.handleSubmit(async (value) => {
+    onSave(value);
+
+    form.reset({
+      experimentEvent: "",
+      experimentIdProperty: "",
+      variationIdProperty: "",
+    });
+  });
+
+  const dataSourceEvents = dataSource.settings.events;
```

Comment:
Because of the bug we just had where `settings` was undefined, maybe we should allow editing here, no matter what so at least they have a way to get out of that state.  The form default value is set to empty strings, so that part should work fine.

Reply:
I've added support for the empty `settings` or `events` here: https://github.com/growthbook/growthbook/pull/550/commits/41cd31afeae20fc8fea220db387268101abcbc55

I've updated the bottom of the PR description with screenshots (same ones I sent on Slack).